File: App.client/projects/proctor-app/scripts/models/case-assignment.ts
import { User, CaseAssignmentStatus } from './user';
import { Complaint } from './complaint';

export class CaseAssignment {
  id: number;
  complaintId: number;
  assignedTo: number;
  assignedBy: number;
  assignedAt: string;
  deadline?: string | null;
  status: string;
  notes?: string | null;
  createdOn: string;
  modifiedOn: string;
  complaint?: Complaint | null;
  assignee?: User | null;
  assigner?: User | null;

  constructor() {
    this.id = 0;
    this.complaintId = this.assignedTo = this.assignedBy = 0;
    this.assignedAt = new Date().toISOString();
    this.deadline = null;
    this.status = CaseAssignmentStatus.Pending;
    this.notes = null;
    this.createdOn = this.modifiedOn = '';
  }
}

File: App.Models/CaseAssignment.cs
using App.Core.Models;

namespace App.Models
{
    public class CaseAssignment : IAuditableEntity
    {
        public int Id { get; set; }
        public int ComplaintId { get; set; }
        public int AssignedTo { get; set; }
        public int AssignedBy { get; set; }
        public DateTime AssignedAt { get; set; }
        public DateTime? Deadline { get; set; }
        public CaseAssignmentStatus Status { get; set; } = CaseAssignmentStatus.Pending;
        public string? Notes { get; set; }
        public DateTime CreatedOn { get; set; }
        public DateTime ModifiedOn { get; set; }

        // Navigation properties
        public virtual Complaint Complaint { get; set; } = null!;
        public virtual User Assignee { get; set; } = null!;
        public virtual User Assigner { get; set; } = null!;
    }
}


